### DISCUSSION TITLE
How to access JDBI from custom Dropwizard Command

### DISCUSSION THREAD

**Discussion by cmsimike** [dropwizard/dropwizard#7231] [Q&A]
Hello!

I am exploring Dropwizard. I'd like to create a command that allows me to create a user from the command line (for you Django users, something like https://docs.djangoproject.com/en/4.2/intro/tutorial02/#creating-an-admin-user).

Following the docs, I'm currently installing an instance of `ConfiguredCommand` in `Application.initialize` but creating the JDBI-related resources in `Application.run`. I don't see anything in the `Bootstrap` class that would allow me to recreate the required object. Currently I'm using this line: `final Jdbi jdbi = factory.build(environment, configuration.getDataSourceFactory(), "postgresql");` however while the `ConfiguredCommand` does have access to the configuration, I am not seeing how I can get the environment.

I'd prefer to use DI to just inject my UserService (which itself makes use of JDBI) but I'm not sure that's possible given I have to create an instance of the `ConfiguredCommand`.

Any tips on how to properly access the database from a `Command` would be helpful. Thanks!

**Comment by joschi** [comment]
You're looking for [`EnvironmentCommand`](https://www.javadoc.io/static/io.dropwizard/dropwizard-core/4.0.0/io/dropwizard/core/cli/EnvironmentCommand.html#run(io.dropwizard.core.setup.Environment,net.sourceforge.argparse4j.inf.Namespace,T)). 😉

**Comment by cmsimike** [comment]
I will check it out, thank you @joschi !
